### PR TITLE
Fix injected script permissions and enhance template creation control

### PR DESCRIPTION
**Pull request by zach-r-long**
Added functionality to set permissions on created template files, fixes serial-startup and makes hostname and timezone scripts executable

### CODE REVIEW

**Comment by activeshadow** [comment]
Hey @zach-r-long, sorry I just saw this PR. What do you think about adding a permissions key to the schema for disk injects, with a sane default if not provided, so permissions can be set there instead of hard-coded again?

**Comment by zach-r-long** [comment]
I agree, that's probably a more complete solution. While we are at it what are your thoughts on owner  user and group as well?

**Comment by activeshadow** [comment]
@zach-r-long I'm open to the schema supporting per-injection user/group settings as well, if you want to take a stab at it. Should probably just default to `root:root` if not specified, since that's what the ownership of injected files gets set to now.

**Comment by zach-r-long** [comment]
Be a couple of weeks till I can get there but sure.

**Comment by zach-r-long** [comment]
Added file permission / owner / group to a specified file path that may be specified. This will require an update to the minimega code base as well which will follow shortly. 
@activeshadow Do you want to separate this as a separate pull request ?

**Comment by activeshadow** [comment]
> Added file permission / owner / group to a specified file path that may be
> specified. This will require an update to the minimega code base as well which
> will follow shortly.
> @activeshadow Do you want to separate this as a separate pull request ?

In this pull request is fine. 👍

**Comment by zach-r-long** [comment]
File permissions are now part of the injection schema / command structure in both minimega and phenix. Suggest the use of UID and GID rather than names. Permissions, recursive, owner, and group are optional parameters. Will default to 0664,root:root if not defined. Removed old way of setting permissions when template created for new way via injection schema.

**Comment by zach-r-long** [comment]
Should be good to go ...

**Review by activeshadow — CHANGES_REQUESTED** [review]
@zach-r-long I think I prefer the limited approach you took in your initial commit where you simply set the file permissions of the files to be injected when you created them. If I recall correctly, when files are injected into a disk, the file permissions are maintained, so that should work fine.

I'm not sure if specifying owner/group details will work very well, given that it would require the owner/group to already exist in the image the file is being injected into. Also, I'm not really wanting to modify any of the actual minimega code since we're trying to target phenix to work with the latest release of minimega (as opposed to the version of minimega that's in this repo). Supporting owner/group settings requires the minimega `disk inject` to be modified (as you've done here) which I'm currently wanting to avoid.

TL;DR Can we go back to your initial commit, then update it to get permissions from the injection struct in the node spec?
